cmd/ethatomicswap/contract/src/contracts/AtomicSwap.sol

```diff
@@ -0,0 +1,187 @@
+// Copyright (c) 2017 Altcoin Exchange, Inc
+
+// Copyright (c) 2018 BetterToken BVBA
+// Use of this source code is governed by an MIT
+// license that can be found at https://github.com/rivine/rivine/blob/master/LICENSE.
+
+pragma solidity ^0.4.23;
+
+// Notes on security warnings:
+//  + block.timestamp is safe to use,
+//    given that our timestamp can tolerate a 30-second drift in time;
```

Comment:
Maybe this is covered somewhere else, but may I ask why this uses timestamps vs using the block number? It seems block times may change, and knowing the number of blocks mined since funds were transferred is really what we're after for security.